docs/src/content/docs/core/define-integration.mdx

```diff
@@ -15,14 +15,21 @@ import {
     defineOptions,
 } from "astro-integration-kit";
 import { corePlugins } from "astro-integration-kit/plugins";
+import { z } from "astro/zod";
 
-type Options = {
-    foo?: string | undefined;
+const OptionsSchema = z.object({
+    foo: z.string().optional().default("bar"),
+});
+
+export type Options = {
+    foo?: z.infer<typeof OptionsSchema>["foo"];
 }
```

Comment:
Why not this?
```suggestion
export type Options = z.input<typeof OptionsSchema>;
```

Reply:
This is to keep JSDoc annotations! I guess I need to add one here so that it makes sense